src/jit/codegenarm64.cpp

```diff
@@ -4601,8 +4601,28 @@ void CodeGen::genSIMDIntrinsicGetItem(GenTreeSIMD* simdNode)
                 assert(op1->isUsedFromReg());
                 regNumber srcReg = op1->gtRegNum;
 
-                // mov targetReg, srcReg[#index]
-                getEmitter()->emitIns_R_R_I(INS_mov, baseTypeSize, targetReg, srcReg, index);
+                instruction ins;
+                if (varTypeIsFloating(baseType))
+                {
+                    assert(genIsValidFloatReg(targetReg));
+                    // dup targetReg, srcReg[#index]
```

Comment:
Why do you change floating logic here? Does your test check this part?

Reply:
Techically, the logic for floating point targetReg hasn't changed.

`emitIns_R_R_I(INS_mov, baseTypeSize, targetReg, srcReg, index)` where targetReg is float register (or vector register) and srcReg is vector register would emit `IF_DV_2E` which is encoding for INS_dup:
https://github.com/dotnet/coreclr/blob/27c848e37e9998142b60e776cf5b5d08a3543fe1/src/jit/emitarm64.cpp#L4504-L4523

I just made the floating point move more explicit by using INS_dup:
https://github.com/dotnet/coreclr/blob/27c848e37e9998142b60e776cf5b5d08a3543fe1/src/jit/emitarm64.cpp#L4748-L4757

But I can add the test cases for `Vector<float>` and `Vector<double>`